forge-gui/res/cardsfolder/upcoming/chaotic_transformation.txt

```diff
@@ -0,0 +1,13 @@
+Name:Chaotic Transformation
+ManaCost:5 R
+Types:Sorcery
+A:SP$ Pump | ValidTgts$ Artifact | TgtPrompt$ Select target artifact | TargetMin$ 0 | TargetMax$ 1 | SubAbility$ DestroyCreature | AILogic$ Destroy | IsCurse$ True | SpellDescription$ Exile up to one target artifact, up to one target creature, up to one target enchantment, up to one target planeswalker, and/or up to one target land. For each permanent exiled this way, its controller reveals cards from the top of their library until they reveal a card that shares a card type with it, puts that card onto the battlefield, then shuffles. | StackDescription$ None
+SVar:DestroyCreature:DB$ Pump | ValidTgts$ Creature | TgtPrompt$ Select target creature | TargetMin$ 0 | TargetMax$ 1 | AILogic$ Destroy | IsCurse$ True | SubAbility$ DestroyEnch | StackDescription$ None
+SVar:DestroyEnch:DB$ Pump | ValidTgts$ Enchantment | TgtPrompt$ Select target enchantment | TargetMin$ 0 | TargetMax$ 1 | AILogic$ Destroy | IsCurse$ True | SubAbility$ DestroyPlaneswalker | StackDescription$ None
+SVar:DestroyPlaneswalker:DB$ Pump | ValidTgts$ Planeswalker | TgtPrompt$ Select target planeswalker | TargetMin$ 0 | TargetMax$ 1 | AILogic$ Destroy | IsCurse$ True | SubAbility$ DestroyLand | StackDescription$ None
+SVar:DestroyLand:DB$ Pump | ValidTgts$ Land | TgtPrompt$ Select target land | TargetMin$ 0 | TargetMax$ 1 | AILogic$ Destroy | IsCurse$ True | SubAbility$ DBDestroy | StackDescription$ None
+SVar:DBDestroy:DB$ ChangeZone | Origin$ Battlefield | Destination$ Exile | Defined$ Targeted | SubAbility$ DBRepeat | RememberLKI$ True
+SVar:DBRepeat:DB$ RepeatEach | DefinedCards$ DirectRemembered | RepeatSubAbility$ DBRevealUntilShare | ClearRemembered$ True | ChooseOrder$ True
```

Comment:
I think this part is probably still incorrect:

> If a player controls multiple permanents that were exiled this way, that player first chooses which target they're considering, then they reveal cards until they reveal a card that shares a type with that permanent as it last existed on the battlefield, then they put that card onto the battlefield and shuffle. They then repeat this process for the remaining targets they controlled.

Might need to extend RepeatEachEffect a bit as the current ChooseOrder implementation isn't sufficient for this. Or you can try also scripting ChooseCard into the loop but might be tricky?

Reply:
Gnarly card!

Sounds like ordering is based on players instead of things exiled, so I think Forge can handle that, it's just the APNAP turn order thing. Only looping players who had a thing exiled, so probably want `RepeatPlayers$` instead of `DefinedCards$`

May need to Imprint on each pump line so you can have players choose from `Card.IsImprinted+RememberedPlayerOwn` as I'm not sure how well targeting is going to last as this gets going.

Not sure I like the `AILogic$ Destroy` as stuff you exile could be replaced by just about anything. It definitely isn't just a destruction card.